packages/netlify-cms-core/src/actions/entries.js

```diff
@@ -452,6 +452,7 @@ export function persistEntry(collection) {
     const state = getState();
     const entryDraft = state.entryDraft;
     const fieldsErrors = entryDraft.get('fieldsErrors');
+    const usedSlugs = selectAllSlugEntries(state, collection.get('name'));
```

Comment:
I could be wrong, but I don't believe `persistEntry` is used when the editorial workflow is active, so selecting just the slugs for published entries should suffice. Did you find that not to be the case?